migration/1643891610952-setSegmentNotifiedForOlderDonations.ts

```diff
@@ -0,0 +1,32 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class setSegmentNotifiedForOlderDonations1643891610952
+  implements MigrationInterface
+{
+  // Update statements don't have if exists clause, we have to run another query or create a pg function
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    const donationTableExists = await queryRunner.query(`
+      SELECT EXISTS (
+        SELECT FROM pg_tables
+        WHERE  schemaname = 'public'
+        AND    tablename  = 'donation'
+      );
+    `);
+    if (donationTableExists[0].exists) {
+      await queryRunner.query(`UPDATE donation SET "segmentNotified" = true`);
+    }
+  }
+
+  public async down(queryRunner: QueryRunner): Promise<void> {
+    const donationTableExists = await queryRunner.query(`
+      SELECT EXISTS (
+        SELECT FROM pg_tables
+        WHERE  schemaname = 'public'
+        AND    tablename  = 'donation'
+      );
+    `);
+    if (donationTableExists[0].exists) {
+      await queryRunner.query(`UPDATE donation SET "segmentNotified" = false`);
+    }
+  }
+}
```

Comment:
I think you can make migration file much easier with this:

```

import { MigrationInterface, QueryRunner } from 'typeorm';

export class setSegmentNotifiedForOlderDonations1643891610952 implements MigrationInterface {
  // Update statements don't have if exists clause, we have to run another query or create a pg function
  async up(queryRunner: QueryRunner): Promise<void> {
    await queryRunner.query(
      `ALTER TABLE IF EXISTS  "donation" ADD  "segmentNotified" boolean DEFAULT true`,
    );
  }

  async down(queryRunner: QueryRunner): Promise<void> {
    await queryRunner.query(
      `ALTER TABLE IF EXISTS  "donation" DROP "segmentNotified"`,
    );
  }
}
```

Other parts of the PR seems good to me

Reply:
The thing is, I don't want to change the default value constraint to **true**, or else my crontask will never run.
Nor drop the column, as the sync mode of typeorm is automatically handling it the creation/dropping of columns.

I only want to set the value of the current donations on the db to true.

Reply:
Let me see if I an try something else

Reply:
Ok, so if you delegate dropping that to typeorm, you can leave the down() function empty, you can change your migration file like this

```
import { MigrationInterface, QueryRunner } from 'typeorm';

export class setSegmentNotifiedForOlderDonations1643891610952 implements MigrationInterface {
  async up(queryRunner: QueryRunner): Promise<void> {
    await queryRunner.query(
      `ALTER TABLE IF EXISTS  "donation" ADD  "segmentNotified" boolean DEFAULT true`,
    );
  }

  async down(queryRunner: QueryRunner): Promise<void> {

  }
}
```
Isn't it enough?

Reply:
Actually you are right, I got confused, we run migrations before the app syncs the database. So if we set it to true in the migration it will work. 
Then the sync shall set it default to false for future donations. 
Also yeah, we can delegate to typeorm the down function.